Blast sequence alignment

azangru opened the pull request:
## Description
This PR adds a simple string representation of an alignment between a query sequence and a match found by the BLAST tool.

**NOTE:** lots of lines in the diff are just sample data in the storybook. We can delete the BlastSequenceAlignment story when we are done with reviewing this PR.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1558

## Deployment URL(s)
~View Storybook locally.~ deleted the storybook story after the review

msmanoj commented on src/content/app/tools/blast/components/blast-sequence-alignment/blastSequenceAlignmentHelper.ts:
```diff
@@ -0,0 +1,165 @@
+
+  for (let i = startIndex; i < endIndex; i++) {
+    if (sequence[i] !== '-') {
+      position += 1 * strandDirectionMultiplier;
```
```suggestion
      position += strandDirectionMultiplier;
```

azangru replied:
I think I explicitly added the `1` for clarity, i.e. "either add or subtract one based on whether you are moving in the same or in the opposite direction as the query sequence". I realise that `1 * strandDirectionMultiplier` looks silly, because it's the same as just `strandDirectionMultiplier`; but then, will it not be confusing that the code would read as if you are adding a multiplier.

msmanoj replied:
Fair enough. Happy to leave it as it is.

msmanoj commented on src/content/app/tools/blast/components/blast-sequence-alignment/blastSequenceAlignmentHelper.ts:
```diff
@@ -0,0 +1,165 @@
+
+const ALIGNMENT_LINE_LENGTH = 60;
+
+type Params = {
```
This is same as `BlastSequenceAlignmentInput`. Can we import and use that instead?

azangru replied:
Yeah, I was planning to do just that, and then forgot 🙂

azangru replied:
✅

azangru commented on src/content/app/tools/blast/types/blastJob.ts:
```diff
@@ -0,0 +1,54 @@
+  hit_hsps: HSP[];
+};
+
+type HSP = {
```
Yes, I git checked-out these files from that branch into this one 🙂  I needed these changes in this branch anyway.

azangru replied:
I git checked-out this file from the branch with the hit diagram, where it was renamed to blastJob:

![image](https://user-images.githubusercontent.com/6834224/184110377-9cfeb4d3-58d5-4363-ba09-5af0b12a268c.png)

Didn't realise the blastApiResponse file would remain unchanged in this branch 😞

azangru replied:
✅

msmanoj commented on src/content/app/tools/blast/components/blast-sequence-alignment/formatters/simpleStringFormatter.ts:
```diff
@@ -0,0 +1,95 @@
+    `${queryLinePrefix} ${queryLine} ${queryLineSuffix}`,
+    `${alignmentPrefix} ${middleLine}`,
+    `${hitLinePrefix} ${hitLine} ${hitLineSuffix}`
+  ].join('\n');
```
Wouldn't it be simpler to automatically align these using a table?

azangru replied:
This is specifically a formatter for generating a raw string output if we need one (at least for testing purposes; but even more importantly, if we ever need to generate download files ourselves). We can create other formatters if we need them, or just generate a table out of the output of the `createBlastSequenceAlignment` function if that is necessary.

msmanoj reviewed (approved):
I am not sure what our plans are regarding the middle line if it needs to count or not.
Other than that, the PR looks good.

Please remember to remove the sample data.

azangru commented:
> I am not sure what our plans are regarding the middle line if it needs to count or not.

I updated the formatter code; it now adds the position in the beginning and the end of the middle line. Plus I changed the labels in the beginning of the line to align right. The alignment looks like this now:

![image](https://user-images.githubusercontent.com/6834224/184127521-8afef2e0-a354-49ee-94f2-07afd7b17d95.png)
